src/lib/net/SecureSocket.cpp

```diff
@@ -675,13 +675,17 @@ bool SecureSocket::verifyCertFingerprint(const deskflow::fs::path &fingerprintDb
   deskflow::openUtf8Path(file, fingerprintDbPath);
   deskflow::FingerprintDatabase db;
   db.read(fingerprintDbPath);
-  if (!db.fingerprints().empty()) {
-    LOG((CLOG_NOTE "read %d fingerprints from %s", db.fingerprints().size(), fingerprintDbPath.c_str()));
-  } else {
+  const bool emptyDB = db.fingerprints().empty();
+
+  if (file.good() && emptyDB) {
     LOG((CLOG_ERR "failed to open trusted fingerprints file: %s", fingerprintDbPath.c_str()));
     return false;
   }
 
+  if (!emptyDB) {
+    LOG((CLOG_NOTE "read %d fingerprints from %s", db.fingerprints().size(), fingerprintDbPath.c_str()));
+  }
```

Comment:
Your latest change makes for confusing reading.

Please use simple early returns instead, like this:
```suggestion
  if (!file.good()) {
    LOG((CLOG_ERR "failed to open trusted fingerprints file: %s", fingerprintDbPath.c_str()));
    return false;
  }

  if (db.fingerprints().empty()) {
    LOG((CLOG_ERR "trusted fingerprints file is empty: %s", fingerprintDbPath.c_str()));
    return false;
  }

  LOG((CLOG_NOTE "read %d fingerprints from: %s", db.fingerprints().size(), fingerprintDbPath.c_str()));
```

Reply:
This is wrong is not an error if we don't have the file that is the reason i changed the code around.

Reply:
if the file exists and its empty thats is a problem 
if the file does not (!file.good()) is not an error we just didn't make it yet
its only a problem if the files exists and is empty

Reply:
I see. In that case, please change the error message to reflect reality:
```cpp
  if (file.good() && emptyDB) {
    LOG((CLOG_ERR "trusted fingerprints file is empty: %s", fingerprintDbPath.c_str()));
    return false;
  }
```

Reply:
if the file is good (i.e it exists) but our database is empty after trying to read that is a read file error. There is no way for this to happen otherwise.

Reply:
OK, understood.